src/Core/Commander/ManagerCommands.js

```diff
@@ -40,6 +40,9 @@ define('Core/Commander/ManagerCommands', [
             this.providerMap = {};
             this.history = null;
             this.eventsManager = new EventsManager();
+            this.currentCommandsCount = 0;
+            this.maxConcurrentCommands = 16;
```

Comment:
I haven't really followed the discussion (and haven't looked at the patch besides those few lines), but to keep a high framerate, the idea is generally to flip a flag whenever there's a need to update the "rendering", and in a `requestAnimationFrame` callback to re-render based on the flag.

I have no idea what commands are and how they're used in iTowns, so I can't really comment on how things should work wrt queuing and executing them, but things I've seen elsewhere related to similarly-named patterns are to:
 * use [`setImmediate`](https://developer.mozilla.org/en/docs/Web/API/Window/setImmediate) or similar to defer processing (see https://www.nczonline.net/blog/2013/07/09/the-case-for-setimmediate/ )
 * run as many commands as possible, timing them until you reach 15ms or so, to keep a 60fps, and deferring the rest to a new `setImmediate` (see https://www.nczonline.net/blog/2009/08/11/timed-array-processing-in-javascript/)

`requestAnimationFrame` should only be used for things that directly affect rendering (so rather as a side-effect of commands IIUC), not as a general-purpose task scheduler.